The DeleteRoute() of OpenStack cloudprovider run panic

**Is this a BUG REPORT or FEATURE REQUEST?**:
 /kind bug

**What happened**:
After starting controller manager, get the following panic:
/----------------------------------
I0712 08:54:30.845526       1 routecontroller.go:175] Deleting route 10.245.1.0/24 10.245.1.0/24
I0712 08:54:30.845548       1 routecontroller.go:146] Creating route for node ab-6l35lzekrg-0-7igjn3vr43tw-kube-minion-abhezrwnqucq 10.245.1.0/24 with hint 4b63c6a0-655c-11e7-9a48-fa163e8338fb, throttled 1.602µs
I0712 08:54:30.846090       1 routecontroller.go:146] Creating route for node ab-hfvphu5tyl-0-623ksgxczeoh-kube-master-oevhfzd3xo7s 10.245.0.0/24 with hint 08eb2c36-655c-11e7-9a48-fa163e8338fb, throttled 651ns
I0712 08:54:31.194807       1 routecontroller.go:154] Created route for node ab-6l35lzekrg-0-7igjn3vr43tw-kube-minion-abhezrwnqucq 10.245.1.0/24 with hint 4b63c6a0-655c-11e7-9a48-fa163e8338fb after 349.278413ms
I0712 08:54:32.429944       1 routecontroller.go:154] Created route for node ab-hfvphu5tyl-0-623ksgxczeoh-kube-master-oevhfzd3xo7s 10.245.0.0/24 with hint 08eb2c36-655c-11e7-9a48-fa163e8338fb after 1.583805322s
panic: runtime error: index out of range

goroutine 1250 [running]:
panic(0x3a3d4e0, 0xc42001a080)
	/usr/local/go/src/runtime/panic.go:500 +0x1a1
k8s.io/kubernetes/pkg/cloudprovider/providers/openstack.(*Routes).DeleteRoute(0xc4212e15a0, 0x40ff60e, 0xa, 0xc421bf22a0, 0x0, 0x0)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/cloudprovider/providers/openstack/openstack_routes.go:265 +0x7d5
k8s.io/kubernetes/pkg/controller/route.(*RouteController).reconcile.func2(0xc4210bed70, 0xc421bf05e0, 0xc421bf22a0, 0xed0f7dac6, 0x3264f461, 0x6f143e0)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/controller/route/routecontroller.go:176 +0x15a
created by k8s.io/kubernetes/pkg/controller/route.(*RouteController).reconcile
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/controller/route/routecontroller.go:183 +0x6e6
/----------------------------------

**What you expected to happen**:
Successed to delete route.

**How to reproduce it (as minimally and precisely as possible)**:
verytime

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): v1.6.4
- Cloud provider or hardware configuration**:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

cc /sig-openstack

--- a/pkg/cloudprovider/providers/openstack/openstack_routes.go
+++ b/pkg/cloudprovider/providers/openstack/openstack_routes.go
@@ -262,8 +262,8 @@ func (r *Routes) DeleteRoute(clusterName string, route *cloudprovider.Route) err
 
 	if index != -1 {
 		// Delete element `index`
-		addr_pairs[index] = addr_pairs[len(routes)-1]
-		addr_pairs = addr_pairs[:len(routes)-1]
+		addr_pairs[index] = addr_pairs[len(addr_pairs)-1]
+		addr_pairs = addr_pairs[:len(addr_pairs)-1]
 
 		unwind, err := updateAllowedAddressPairs(r.network, &port, addr_pairs)
 		if err != nil {

